eth-state-trie/test/resolver.spec.js

```diff
@@ -7,185 +7,131 @@ const expect = chai.expect
-    stateRoot: new Buffer('012304a817c80004a817c80004a817c80004a817c80004a817c80004a817c800', 'hex')
-  }
-  function prepareTestContract (done) {
-    testContract = new Account(testContractData)
-    done()
+    balance: Buffer.from('012345', 'hex'),
+    codeHash: Buffer.from('abcd04a817c80004a817c80004a817c80004a817c80004a817c80004a817c800', 'hex'),
+    stateRoot: Buffer.from('012304a817c80004a817c80004a817c80004a817c80004a817c80004a817c800', 'hex')
   }
 
   // setup external account test data
   let testExternalAccount
   let testExternalAccountData = {
-    balance: new Buffer('abcdef', 'hex'),
-    nonce: new Buffer('02', 'hex')
-  }
-  function prepareTestExternalAccount (done) {
-    testExternalAccount = new Account(testExternalAccountData)
-    done()
+    balance: Buffer.from('abcdef', 'hex'),
+    nonce: Buffer.from('02', 'hex')
   }
 
   // setup test trie
   let trie
   let trieNodes = {}
   let dagNodes = {}
-  before((done) => {
+  before(async () => {
     trie = new Trie()
-    async.waterfall([
-      (cb) => prepareTestContract(cb),
-      (cb) => prepareTestExternalAccount(cb),
-      (cb) => populateTrie(trie, cb),
-      (cb) => dumpTrieNonInlineNodes(trie, trieNodes, cb),
-      // (cb) => logTrie(trie, cb),
-      (cb) => async.mapValues(trieNodes, (node, key, cb) => {
-        ipldEthStateTrie.util.serialize(node, cb)
-      }, cb)
-    ], (err, result) => {
-      if (err) {
-        return done(err)
-      }
-      dagNodes = result
-      done()
+    testContract = new Account(testContractData)
+    testExternalAccount = new Account(testExternalAccountData)
+    await populateTrie(trie)
+    await dumpTrieNonInlineNodes(trie, trieNodes)
+    Object.entries(trieNodes).map(([key, node]) => {
+      dagNodes[key] = ipldEthStateTrie.util.serialize(node)
     })
   })
 
-  function populateTrie (trie, cb) {
-    async.series([
-      (cb) => trie.put(new Buffer('000a0a00', 'hex'), testExternalAccount.serialize(), cb),
-      (cb) => trie.put(new Buffer('000a0a01', 'hex'), testExternalAccount.serialize(), cb),
-      (cb) => trie.put(new Buffer('000a0a02', 'hex'), testExternalAccount.serialize(), cb),
-      (cb) => trie.put(new Buffer('000a0b00', 'hex'), testExternalAccount.serialize(), cb),
-      (cb) => trie.put(new Buffer('000b0a00', 'hex'), testContract.serialize(), cb),
-      (cb) => trie.put(new Buffer('000b0b00', 'hex'), testContract.serialize(), cb),
-      (cb) => trie.put(new Buffer('000c0a00', 'hex'), testContract.serialize(), cb)
-    ], (err) => {
-      if (err) return cb(err)
-      cb()
-    })
-  }
-
-  function cid (data, cb) {
-    multihashing(data, 'keccak-256', (err, hash) => {
-      if (err) {
-        return cb(err)
-      }
-      const cid = new CID(1, resolver.multicodec, hash)
-      cb(null, cid)
+  function populateTrie (trie) {
+    return new Promise((resolve, reject) => {
+      async.series([
```

Comment:
you could just `let put = promisify(trie.put.bind(trie))` and then `await put(...)` of these calls, then async can be removed as a dependency